### DISCUSSION TITLE
Changing parameters while the pump is operating

### DISCUSSION THREAD

**Discussion by krzbor** [Egyras/HeishaMon#566] [Q&A]
In the K-series heat pump, changing the parameters of the heating curve and delta requires turning the pump off. Using HeishaMon, however, I can make these changes while the pump is running.

I've noticed, though, that the changes are not correctly interpreted. Even worse, the pump doesn't operate according to either the previous settings or the new ones.

What are your experiences? Is changing parameters that normally require the pump to be turned off correct and permissible? Or should the pump always be turned off before changing the parameters?

**Comment by geduxas** [comment]
Settings from cloud also don't turn pump off, so i don't think ir should. Also maybe some settings need's for restart? Didn't notice any setting not applied for H series pump.

What setting for you is not working?

**Reply by krzbor** [reply]
It's about the heating curve settings. My main question is whether such parameters can be adjusted while the pump is running. As I mentioned, to change the heating curve on the display, you need to access the service settings. However, the pump does not allow this operation and requires turning it off first.

**Reply by geduxas** [reply]
Yes, from pump it's required to stop. Truly don't know why, but if you are doing it pretty fast, pump will not actually stop. Also doing same from cloud, pump is not stopping... 

So do you really tested and it's not working or just trying to understand?
For H/J series there's no such problem, also node red heishamon integration have heating curve expansion with 3 points (originally it's only 2), so it's dynamically changes curve in desired setup.

**Comment by MiG-41** [comment]
Changing the delta during compressor runing at my H for sure not is effective everytime. 

But changing the curves for sure influence on target temp on fly.

**Comment by krzbor** [comment]
In my case, the pump was on but not actively running at that moment. I changed the heating curve, but when the temperature reached the point where it should have activated, the compressor did not start.

I turned the pump off and on again—this time, the compressor started up.

This made me wonder: is it acceptable to change these parameters while the pump is on, considering that when I try to make these changes on the pump display, I’m required to turn it off first?

**Reply by geduxas** [reply]
It could be something new to K/L series. It's still pretty new devices and need's more tests. Do you have original TAW?  You could do same test's from cloud.

**Comment by MiG-41** [comment]
This is becouse turning on compressor again is something different then target temperature. 
Please see here: https://github.com/Egyras/HeishaMon/discussions/562
Switching on again the compressor has nothing to do with Target , only with Inlet and Outlet temp during stop...
